### PR TITLE
Version 1.1.2

### PR DESCRIPTION
**Pull request by JosunLP**
This pull request focuses on improving link security in the HTML sanitizer by automatically adding `rel="noopener noreferrer"` to external and `target="_blank"` links, updating documentation and tests accordingly, and making minor documentation and test improvements. These changes help prevent tabnabbing and referrer leakage vulnerabilities while ensuring the sanitizer's behavior is well-documented and robustly tested.

**Security enhancements:**

* Added logic in `sanitize.ts` to automatically add `rel="noopener noreferrer"` to `<a>` tags that either have `target="_blank"` or point to external domains, including protocol-relative URLs and special protocols like `mailto:` and `tel:`. Existing `rel` values are preserved and merged. [[1]](diffhunk://#diff-661a9f5ca84cb3668a2ebf96d24a7724d789597aad92ad19dc2ac9019b36e848R482-R503) [[2]](diffhunk://#diff-661a9f5ca84cb3668a2ebf96d24a7724d789597aad92ad19dc2ac9019b36e848R356-R399)
* Updated the set of allowed attributes in the sanitizer to include `rel` and `target`.

**Documentation updates:**

* Updated `CHANGELOG.md` to document the security improvements and documentation fixes.
* Expanded the security guide (`docs/guide/security.md`) with a new section on automatic link security, including code examples for various link scenarios.
* Updated usage examples in `README.md` to demonstrate the new link security behavior.
* Fixed import paths and improved error handling in the agents guide, and clarified backend call error handling. [[1]](diffhunk://#diff-26313617f907237bd6d0bcea48cfd9ff8ff54c51f38731fc693e5ceaa8a93fe1L56-R57) [[2]](diffhunk://#diff-26313617f907237bd6d0bcea48cfd9ff8ff54c51f38731fc693e5ceaa8a93fe1R66-R108) [[3]](diffhunk://#diff-26313617f907237bd6d0bcea48cfd9ff8ff54c51f38731fc693e5ceaa8a93fe1L112-R139)

**Testing improvements:**

* Added comprehensive tests to `security.test.ts` to cover the new link security behavior, including edge cases for various protocols, casing, whitespace, SSR/Node.js environments, and preservation of existing `rel` values.
* Minor improvements to utility tests for reliability and coverage. [[1]](diffhunk://#diff-c37842b08434952211c06ad3cdc71258303dcce7491a2b5f2005876cfbc26681L134-R134) [[2]](diffhunk://#diff-c37842b08434952211c06ad3cdc71258303dcce7491a2b5f2005876cfbc26681R196-R198)

**Version bump:**

* Bumped package version to `1.1.2` in `package.json` to reflect the new release.

### CODE REVIEW

File: src/security/sanitize.ts [hunk 1]
```diff
@@ -351,6 +353,50 @@ const isSafeUrl = (value: string): boolean => {
...
+      return false;
+    }
+    
+    // In non-browser environments (e.g., Node.js), treat all absolute URLs as external
+    if (typeof window === 'undefined' || !window.location) {
+      return true;
+    }
+    
+    const urlObj = new URL(trimmedUrl, window.location.href);
+    return urlObj.origin !== window.location.origin;
```
[the remark this answers is not in the document]
Comment (JosunLP) [reply src/security/sanitize.ts:397]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/bQuery/bQuery/pull/12#discussion_r2723380727)
